Userland/Libraries/LibC/fcntl.cpp

```diff
@@ -128,7 +128,12 @@ int utimensat(int dirfd, char const* path, struct timespec const times[2], int f
         errno = EFAULT;
         return -1;
     }
-    return __utimens(dirfd, path, times, flag);
+
+    // Ignore the fd argument if we have an absolute path.
+    if (path[0] == '/')
+        return __utimens(AT_FDCWD, path, times, flag);
+    else
+        return __utimens(dirfd, path, times, flag);
```

Comment:
What was it doing instead? Shouldn't this check be in the kernel?

Reply:
> It was trying to resolve the path against whatever fd was passed to this function, which breaks since libfuse just passes 0 as it expects that the fd would get ignored given that the provided path is absolute.

Shouldn't libfuse be able expect that? In what case does the passed fd matter if the to-be-resolved path is absolute, it would still result in the same absolute path?

Is the root of the issue we are trying to solve here that we are passing a file fd to `custody_for_dirfd`, which then errors out with `ENOTDIR`?

In any case, it feels like we should find better ways to handle this than fixing the issue one-by-one for each user of `custody_for_dirfd`.

Reply:
Oh, I see, you are also thinking about all the other error cases. I was thinking about the `ENOTDIR` in particular.

Still, this situation really isn't architecturally sound, nor is papering over it in a single syscall or every single syscall. The "problem" goes all the way down to `VirtualFileSystem::resolve_path_without_veil`, which wants a `NonnullRefPtr<Custody>`, but won't even look at it if `path[0] == '/'` (which is the same check that you want to do at the syscall layer instead of the VFS layer) and discards it in favor of the root custody.

The "cleanest" solution would probably be to pass a `Function<ErrorOr<NonnullRefPtr<Custody>>()>` instead, and to only call that on demand.

Reply:
The currently linked commit doesn't seem to build, but in either case, this seems to require more engineering effort at another point in time. Let's go with the band aid for now to not unnecessarily stall the FUSE work, but this deserves a large mental FIXME before we do this for any other syscall.